File: Tutorial5-3bDagger-Android/src/main/java/com/example/tutorial5_1daggerandroid/MyApplication.java
package com.example.tutorial5_1daggerandroid;

import com.example.tutorial5_1daggerandroid.di.DaggerApplicationComponent;

import dagger.android.AndroidInjector;

public class MyApplication extends dagger.android.support.DaggerApplication {

    @Override
    protected AndroidInjector<? extends dagger.android.support.DaggerApplication> applicationInjector() {
        return DaggerApplicationComponent.create();
    }
}

File: Tutorial6-4ComponentFactoryDagger-Android-AAC/src/main/java/com/smarttoolfactory/tutorial6_4componentfactorydagger_android/MyApplication.kt
package com.smarttoolfactory.tutorial6_4componentfactorydagger_android

import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.AppComponent
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

class MyApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        val appComponent: AppComponent = DaggerAppComponent.factory().create(this)
        appComponent.inject(this)
        return appComponent
    }
}